components/butterfly/src/rumor/dat_file.rs

```diff
@@ -368,8 +368,14 @@ impl DatFile {
             .read()
             .expect("Member list lock poisoned");
         for member in members.values() {
-            if let Some(membership) = member_list.membership_for(&member.id) {
-                total += self.write_member(writer, &membership)?;
+            if let Some(health) = member_list.health_of(member) {
```

Comment:
I think some comments here stating why we're using this, rather than `member_list.membership_for` will be useful, so some zealous engineer doesn't refactor this fix away sometime in the future.

Reply:
totally! I'm reading docs on the underlying lock implementations on windows in hopes to make a coherent comment.